### PR TITLE
Standardize help output

### LINKED ISSUE #852: Standardize help output
The standard for command line interface help outputs is to print the help message without any newlines before or after the message. Currently, Commander does both.

Commander should better adhere to the standards set by other command line interfaces like `node`, `git`, `bash`,  `grep`, etc.

Changes Needed
- [x] Remove newline before help output
- [x] Remove newline after help output
- [x] Update documentation (particularly the "Custom help" section)
- [x] Update help output tests

Related to #832.

### PR DESCRIPTION
**Pull request by jaredpetersen**
Cleans up the help output by:
- removing the newline before and after
- un-indenting commands and options

Fixes #852 and relates to #832.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -580,7 +580,7 @@ Command.prototype.executeSubCommand = function(argv, args, unknown) {
   proc.on('close', process.exit.bind(process));
   proc.on('error', function(err) {
     if (err.code === 'ENOENT') {
-      console.error('\n  %s(1) does not exist, try --help\n', bin);
+      console.error('%s(1) does not exist, try --help', bin);
     } else if (err.code === 'EACCES') {
       console.error('\n  %s(1) not executable. try chmod or run with root\n', bin);
```
Comment (vanesyan) [inline index.js]: Can you also remove `\n` for `EACCES` error as you did it for `ENOENT`

Comment (jaredpetersen) [reply index.js]: Fixed in the latest update.

**Comment by jaredpetersen** [comment]
@vanesyan Updated PR with your suggested changes. Good catch 👍

**Comment by jaredpetersen** [comment]
@vanesyan When will this be released?

**Comment by jaredpetersen** [comment]
@vanesyan gentle bump

**Comment by abetomo** [comment]
We released 2.18.0.

**Comment by jaredpetersen** [comment]
Thanks!

**Comment by mojavelinux** [comment]
Another thing I've noticed is that commander.js adds a blank line after the category (such as options and commands). Most commands I use don't include this extra line.

commander.js:

```
Options:

  -v, --version                  output the version number
```

other commands

```
Options:
  -v, --version                  output the version number
```

Separate issue?

**Comment by jaredpetersen** [comment]
@mojavelinux I think a separate issue would be good.

**Comment by jaredpetersen** [comment]
@mojavelinux Just created one: #863.

**Comment by mojavelinux** [comment]
Thanks a lot for taking care of that, @jaredpetersen. Much appreciated!

**Pull request merged**